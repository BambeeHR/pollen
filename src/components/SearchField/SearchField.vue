<template>
  <label class="search-field" :class="{ rounded }">
    <BaseIcon class="search-field__icon" icon="search" />
    <span class="search-field__label">{{ placeholder }}</span>
    <input
      :value="value"
      :placeholder="placeholder"
      class="search-field__input"
      type="search"
      v-bind="$attrs"
      @input="$emit('input', $event.target.value)"
    />
  </label>
</template>

<script>
import BaseIcon from '../BaseIcon/BaseIcon.vue';

export default {
  components: { BaseIcon },
  inheritAttrs: false,
  props: {
    placeholder: {
      type: String,
      default: 'Search',
    },
    value: {
      type: String,
      default: '',
    },
  },
};
</script>

<style scoped>
.search-field {
  @apply block
    rounded-md
    relative
    overflow-hidden
    w-full;
}

.search-field__icon {
  @apply absolute 
    left-0 
    ml-4
    text-gray-400
    text-2xl
    top-1/2
    transform
    -translate-y-1/2;
}

.search-field__label {
  @apply sr-only;
}

.search-field__input {
  @apply bg-gray-200
    block
    pl-12
    pr-4
    py-3
    w-full;
}

.search-field__input::placeholder {
  @apply text-gray-300;
}
</style>
