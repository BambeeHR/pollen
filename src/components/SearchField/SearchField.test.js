import { shallowMount } from '@vue/test-utils';
import SearchField from './SearchField.vue';

describe('SearchField', () => {
  test('renders correctly', () => {
    const wrapper = shallowMount(SearchField, {
      propsData: {
        value: 'test',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
