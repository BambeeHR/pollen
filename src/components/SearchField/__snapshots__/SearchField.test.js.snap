// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField renders correctly 1`] = `
<label
  class="search-field"
>
  <baseicon-stub
    class="search-field__icon"
    icon="search"
  />
   
  <span
    class="search-field__label"
  >
    Search
  </span>
   
  <input
    class="search-field__input"
    placeholder="Search"
    type="search"
  />
</label>
`;
